### DISCUSSION TITLE
Enhancement: Support for Nested Markdown Blocks

### DISCUSSION THREAD

**Discussion by FabianHertwig** [danny-avila/LibreChat#3512] [Feature Requests & Suggestions]
### What happened?

When there is a Markdown block inside a markdown block, then the rendering of the outer markdown block ends when the inner markdown block ends with three backticks.

I would expect that the outer Markdown block is rendered until it is closed again.

### Steps to Reproduce

Instruct a model to create such a Markdown structure: `Put out a markdown block that contains another markdown block with some lorum ipsum text before and after each`

The unrendered output is the following:
    
    Sure! Here's how you can structure a markdown block containing another markdown block with Lorem Ipsum text before and after:
    
    ````markdown
    Here is some text before the inner markdown block.
    
    ````markdown
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
    Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
    ```` <!-- rendering ends here, see screenshot -->
    
    Here is some text after the inner markdown block.
    ````
    
    You can copy and paste this code directly into any markdown editor, and it will render as intended!

### What browsers are you seeing the problem on?

Safari

### Screenshots

<img width="807" alt="image" src="https://github.com/user-attachments/assets/b2ac8013-cd72-4713-b043-9b63e90de0a4">

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by berry-13** [comment]
not a bug but rather a feature request. ChatGPT act the same as LibreChat does

![image](https://github.com/user-attachments/assets/6106dc23-10ab-4ac2-8e59-051850c746c8)
